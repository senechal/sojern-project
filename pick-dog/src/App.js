import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from 'react-router-dom';
import { NotificationContainer } from 'react-notifications';
import Fav from './components/Fav';
import Pick from './components/Pick';
import { Nav } from './components/styles';

function App() {
  return (
    <>
      <Router>
        <div>
          <header>
            <Nav>
              <Link to="/" >Home</Link> | <Link to="/favorites" >Favorites</Link>
            </Nav>
          </header>
        </div>
        <Switch>
            <Route path="/favorites">
              <Fav />
            </Route>
            <Route path="/">
              <Pick />
            </Route>
          </Switch>
      </Router>
      <NotificationContainer />
    </>
  );
}

export default App;
