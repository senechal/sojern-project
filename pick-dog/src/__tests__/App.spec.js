import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

import App from '../App';

jest.mock('../components/Pick');
jest.mock('../components/Fav');
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  BrowserRouter: ({children}) => <div>{children}</div>,
}));


describe('App tests', () => {

  it('should render Home Page correctly', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render Favorites Page correctly', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/favorites']}>
        <App />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
