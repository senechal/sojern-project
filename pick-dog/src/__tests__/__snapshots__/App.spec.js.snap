// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render Favorites Page correctly 1`] = `
<div>
  <div>
    <div>
      <header>
        <nav
          class="sc-dlfnbm VWpaN"
        >
          <a
            href="/"
          >
            Home
          </a>
           | 
          <a
            href="/favorites"
          >
            Favorites
          </a>
        </nav>
      </header>
    </div>
    <div>
      Fav Mock
    </div>
  </div>
  <div
    class="notification-container notification-container-empty"
  >
    <div />
  </div>
</div>
`;

exports[`App tests should render Home Page correctly 1`] = `
<div>
  <div>
    <div>
      <header>
        <nav
          class="sc-dlfnbm VWpaN"
        >
          <a
            href="/"
          >
            Home
          </a>
           | 
          <a
            href="/favorites"
          >
            Favorites
          </a>
        </nav>
      </header>
    </div>
    <div>
      Pick Mock
    </div>
  </div>
  <div
    class="notification-container notification-container-empty"
  >
    <div />
  </div>
</div>
`;
